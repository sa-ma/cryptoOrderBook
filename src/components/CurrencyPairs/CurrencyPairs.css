.pair-header {
  width: 100%;
  background-color: var(--secondary-color);
  padding: 2rem;
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
}
.pair-header label {
  color: white;
  font-weight: bold;
  margin: 0 20px;
  font-size: 1.2em;
}
.pair-header select {
  background: transparent;
  border: 1px solid white;
  border-radius: 3px;
  padding: 10px;
  color: white;
  font-size: 1em;
}
.pair-header select:focus {
  outline: none;
}
.pair-header select option {
  color: var(--secondary-color);
  cursor: pointer;
}
.pair-header select option:hover {
  color: white;
  background-color: var(--secondary-color);
}
