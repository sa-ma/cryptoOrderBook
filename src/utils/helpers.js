export const subscribeAction = (action, channel) => ({
  event: `bts:${action}`,
  data: {
    channel: `order_book_${channel}`
  }
});

export const getCurrency = (pair) => {
  return pair.toUpperCase().match(/.{1,3}/g);
};
