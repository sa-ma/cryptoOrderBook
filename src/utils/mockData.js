export const initialState = {
  loading: false,
  currencyPairs: [],
  error: {}
};

export const getCurrencyData = {
  data: [
    {
      base_decimals: 8,
      minimum_order: '25.0 USD',
      name: 'LTC/USD',
      counter_decimals: 2,
      trading: 'Enabled',
      url_symbol: 'ltcusd',
      description: 'Litecoin / U.S. dollar'
    }
  ]
};
export const getErrorMessage = 'Request failed with status code 500';
