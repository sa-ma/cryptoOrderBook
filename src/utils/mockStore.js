import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);

const makeMockStore = (state = {}) => {
  return mockStore({
    ...state
  });
};

export default makeMockStore;
