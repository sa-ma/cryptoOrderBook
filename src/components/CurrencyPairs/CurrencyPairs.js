import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchCurrencyPair } from '../../redux/actions/currencyPairAction';
import { currencySubscriptionAction } from '../../redux/actions/currencySubscriptionAction';
import './CurrencyPairs.css';

const CurrencyPairs = ({
  currencyPairs,
  fetchCurrencyPair,
  currencySubscriptionAction
}) => {
  useEffect(() => {
    fetchCurrencyPair();
  }, [fetchCurrencyPair]);

  const handleChange = (event) => {
    const index = event.target.selectedIndex - 1;
    const singleCurrency = currencyPairs[index];
    currencySubscriptionAction(singleCurrency);
  };
  const PairList = () =>
    currencyPairs.map(({ name }, index) => (
      <option key={index} name={index}>
        {name}
      </option>
    ));

  return (
    <div className="pair-header">
      <label>Related Pairs</label>
      <select
        name="pairList"
        defaultValue="Choose Pair"
        onChange={handleChange}
      >
        <option value="Choose Pair" disabled>
          Choose Pair
        </option>
        <PairList />
      </select>
    </div>
  );
};

CurrencyPairs.propTypes = {
  currencyPairs: PropTypes.array.isRequired,
  fetchCurrencyPair: PropTypes.func.isRequired,
  currencySubscriptionAction: PropTypes.func
};

const mapStateToProps = ({ currencyPairs }) => ({
  currencyPairs: currencyPairs.currencyPairs
});

export const CurrencyPairsComponent = CurrencyPairs;

export default connect(
  mapStateToProps,
  { fetchCurrencyPair, currencySubscriptionAction }
)(CurrencyPairs);
