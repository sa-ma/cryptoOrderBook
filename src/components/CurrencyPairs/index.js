import CurrencyPairs from './CurrencyPairs';

export default CurrencyPairs;
