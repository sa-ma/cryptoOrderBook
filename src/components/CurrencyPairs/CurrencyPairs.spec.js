import React from 'react';
import { mount } from 'enzyme';
import makeMockStore from '../../utils/mockStore';
import { initialState, getCurrencyData } from '../../utils/mockData';
import { CurrencyPairsComponent } from './CurrencyPairs';

const store = makeMockStore(initialState);

describe('<CurrencyPair/> test', () => {
  it('should render component without crashing', () => {
    const props = {
      ...initialState,
      fetchCurrencyPair: jest.fn()
    };
    const component = mount(
      <CurrencyPairsComponent store={store} {...props} />
    );
    expect(component).toMatchSnapshot();
  });
  it('should render a list of currencies after loading successfully', () => {
    const props = {
      currencyPairs: [getCurrencyData],
      fetchCurrencyPair: jest.fn()
    };
    const component = mount(
      <CurrencyPairsComponent store={store} {...props} />
    );
    expect(component.find('div').hasClass('pair-header')).toBe(true);
    expect(component.find('select')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(1);
  });
  it('should call onChange for currency change', () => {
    const props = {
      currencyPairs: [getCurrencyData],
      fetchCurrencyPair: jest.fn(),
      currencySubscriptionAction: jest.fn()
    };
    const handleChange = jest.fn();
    const component = mount(
      <CurrencyPairsComponent
        onChange={handleChange}
        store={store}
        {...props}
      />
    );
    const currencyList = component.find('select');
    currencyList.simulate('change', {
      target: {
        name: 'Choose Pair',
        value: 'Choose Pair'
      }
    });
    expect(currencyList).toHaveLength(1);
    expect(currencyList.html()).toMatch('Choose Pair');
  });
});
