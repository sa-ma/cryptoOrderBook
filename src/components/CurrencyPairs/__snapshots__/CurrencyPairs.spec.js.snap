// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyPair/> test should render component without crashing 1`] = `ReactWrapper {}`;
